src/background/services/web3/handlers/connect.ts

```diff
@@ -17,8 +22,15 @@ import { openApprovalWindow } from '@src/background/runtime/openApprovalWindow';
  * @returns
  */
 
+type Params =
+  | {
+      addressVM?: NetworkVMType;
+      onlyIfTrusted?: boolean;
+    }
+  | undefined;
```

Comment:
I might be missing something but I can't find this. Where are these params coming from?

Reply:
This is sent by the provider for Solana (see [vm module PR](https://github.com/ava-labs/vm-modules/pull/254) - I'll be cleaning it up & adding tests tomorrow).

Reply:
Might also be a better idea to create a new handler for it. WDYT?

Reply:
Alright, I reverted this particular handler to not accept any params again, it simply defaults to `EVM` when adding permissions: f462de8fe.

I'll move the logic for `DAppProviderRequest.WALLET_CONNECT` to an external file in this PR:
* https://github.com/ava-labs/core-extension/pull/170

Reply:
Done here: https://github.com/ava-labs/core-extension/pull/170/commits/55b3c4062881de231d58c624961b26d07912293d